.circleci/config.yml

```diff
@@ -60,9 +60,22 @@ _commands:
               sha256sum $PWD/checksum.txt >> checksum.txt
               apt-get update
               rosdep update
+              
+              # workarround for OMPL and rosdep
+              # https://github.com/ompl/ompl/issues/753
+              # Prevent searching $ROS_WS/install given it's too big for rosdep
+
+              if [ "$ROS_WS" == "<< parameters.underlay >>" ]; then
+                underlay_ws=""
+              else
+                underlay_ws=<< parameters.underlay >>/src
+              fi
+              echo underlay_ws = $underlay_ws
+
               dependencies=$(
                 rosdep install -q -y \
                   --from-paths src \
+                    $underlay_ws \
```

Comment:
To make sure I understand this, this is saying if we're in the overlay workspace, include the underlay workspace in the rosdep install command? Is that because when it runs with both sources, it combines them and views this as within the same list?

Reply:
Yep. The `--from-paths` is to specify a list of paths to recursively search for `package.xml` files. With `pwd` being the overlay workspace, we just add the absolute path to the underlay source folder to ensure we find any packages that are missing from the ament index we sourced:

https://github.com/ros-planning/navigation2/blob/5d3c7e33d2a739acd5eb4ac7e5e4253aa643c96c/.circleci/config.yml#L56

This patch isn't very formulaic, as it can't really recursively scale, e.g. if a package fails to install itself correctly in your underlay's underlay, your SOL; vs. ament index that recursively daisy chains for us.

Recursively searching paths in python is also slow, and seems to stall on the ros install folder.